spk/ffsync/src/wizard/install_uifile.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+INTERNAL_IP=$(ip -4 route get 8.8.8.8 | awk '/8.8.8.8/ && /src/ {print $NF}')
+
+quote_json ()
+{
+    sed -e 's|\\|\\\\|g' -e 's|\"|\\\"|g'
+}
+
+page_append ()
+{
+    if [ -z "$1" ]; then
+        echo "$2"
+    elif [ -z "$2" ]; then
+        echo "$1"
+    else
+        echo "$1,$2"
+    fi
+}
+
+checkPublicUrl()
+{
+    CHECK_PUBLIC_URL=$(/bin/cat<<EOF
+{
+    var publicUrl = arguments[0];
+    var step = arguments[2];
+    var clientUrl = step.getComponent("client_ffsync_public_url");
+    var ipRegex = /^http:\/\/(\d{1,3}\.){3}\d{1,3}:8132$/;
+    var domainRegex = /^http:\/\/((?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}):8132$/;
+    if (ipRegex.test(publicUrl) || domainRegex.test(publicUrl)) {
+        clientUrl.setReadOnly(false);
+        clientUrl.setValue(publicUrl + "/token/1.0/sync/1.5");
+        clientUrl.setReadOnly(true);
+    } else {
+        return "Invalid URL format. Please provide a valid URL.";
+    }
+    return true;
+}
+EOF
+)
+    echo "$CHECK_PUBLIC_URL" | quote_json
+}
+
+PAGE_FFSYNC_SETUP=$(/bin/cat<<EOF
+{
+    "step_title": "Firefox Sync Server 1.5 database configuration",
+    "invalid_next_disabled_v2": true,
+    "items": [{
+        "type": "password",
+        "desc": "Enter your MySQL root password",
+        "subitems": [{
+            "key": "wizard_mysql_password_root",
+            "desc": "Root password",
+            "validator": {
+                "allowBlank": false
+            }
+        }]
+    }]
+}, {
+    "step_title": "Public URL",
+    "invalid_next_disabled_v2": true,
+    "items": [{
+        "type": "textfield",
+        "desc": "Please provide the client-visible URL. If utilizing a domain, ensure it follows the format http://hostname.domain:8132",
+        "subitems": [{
+            "key": "wizard_ffsync_public_url",
+            "desc": "Public URL",
+            "defaultValue": "http://${INTERNAL_IP}:8132",
+            "validator": {
+                "fn": "$(checkPublicUrl)"
+            }
```

Comment:
There exists a predefined validator for url (and for email, alpha and alphanum)
```suggestion
            "validator": {
                "vtype": "url",
                "allowBlank": false
            }
```

Reply:
based on the developer guide, their definition of url will not work for us. their example in the document has this:

> The following is an example of value setting for your URL of the application:
> “url”: http://www.synology.com/
> “url”: “3rdparty/xxx/index.html”